docs/rfc/0002-plugins.md

```diff
@@ -0,0 +1,184 @@
+# Plugins
+
+## Summary
+
+We have found a number of use cases where we'd like to extend the
+functionality of `jk` but don't want to link the core runtime w specific
+libraries.
+
+With a plugin system, we could implement these features outside of the
+runtime with and provide simple way to extend `jk` and experiment with ideas.
+
+The proposal is to:
+
+- Use [go-plugin](https://github.com/hashicorp/go-plugin), the plugin system
+underpinning terraform.
+- Define a first integration point to extend `jk` functionality: `Render`
+plugins.
+- Implement a helm plugin in `@jkcfg/kubernetes` that renders helm charts and
+make its Kubernetes objects available to the `jk` runtime for further
+manipulation.
+
+## Example
+
+1. A [helm](https://helm.sh/) `Renderer` plugin would render a helm chart
+from values specified in `js` and return an array of Kubernetes objects.
+
+1. A Dockerfile `Validator` plugin could validate Dockerfiles that we
+write.
+
+1. An [Open Policy Agent](https://www.openpolicyagent.org/) `Validator`
+plugin would ensure a set of configuration files passes a policy written in
+[Rego](https://www.openpolicyagent.org/docs/latest/policy-language/)
+
+## Motivation
```

Comment:
The motivation is well stated :+1:

Reply:
OPA is another great example -- it's a stretch to include OPA code in the runtime, but it makes total sense for it to be available _somehow_.